pkg/controllers/dynakube/extension/reconciler.go

```diff
@@ -30,11 +30,18 @@ func NewReconciler(clt client.Client, apiReader client.Reader, dynakube *dynatra
 	}
 }
 
-func (r *reconciler) Reconcile(_ context.Context) error {
+func (r *reconciler) Reconcile(ctx context.Context) error {
+	log.Info("start reconciling extensions")
+
 	if r.dynakube.PrometheusEnabled() {
-		log.Info("reconcile extensions")
+		err := reconcileSecret(ctx, r.dynakube, r.client, r.apiReader)
+		if err != nil {
+			setSecretCreatedFalse(r.dynakube.Conditions(), err)
+
+			return err
+		}
 
-		return nil
+		setSecretCreatedTrue(r.dynakube.Conditions())
```

Comment:
the `status` field of dynakubeV1beta3 instance is updated
```
	extensionReconciler := extension.NewReconciler(controller.client, controller.apiReader, dynakubeV1beta3)
```

Reply:
[Fixed](https://github.com/Dynatrace/dynatrace-operator/pull/3426/commits/8335f7bc3ebed57836edac3177649bc6f516529a)